### ISSUE TITLE
Obs overlay timing/value/site disappearance problem when stepping through time

### ISSUE THREAD

**Issue by jherwehe** [CEMPD/VERDI#209]
This bug was first mentioned in a May 22, 2019, comment as an additional problem in Issue #201, but probably deserves its own issue.

Working interactively using VERDI_2.0_beta_linux64_20190623.tar.gz here.  The Linux version allows the binary observation data file to be loaded (unlike the current Windows version of VERDI).  The initial loading of the obs overlay appears to display correctly for the current displayed time, but using the Time Step spinner to change the time (either forward or backward) drastically changes the obs values or, if starting at 00UTC in the example , causes most of the obs sites to disappear at 01UTC and then gradually reappear as you advance the time step, always showing what appear to be incorrect obs values.  Here are a couple of examples at different times showing a good overlay display on the left and incorrect overlay display on the right:

![obs_overlay_problem_comparison_02UTC](https://user-images.githubusercontent.com/16904596/60530517-e7ec6e00-9cc6-11e9-80c5-68348bfecceb.png)
![obs_overlay_problem_comparison_20UTC](https://user-images.githubusercontent.com/16904596/60530529-ef137c00-9cc6-11e9-9829-34b341f46ae3.png)

Please examine the following PDF for easier detailed viewing of additional examples of this problem at different times:
[obs_overlay_problem_comparison.pdf](https://github.com/CEMPD/VERDI/files/3351086/obs_overlay_problem_comparison.pdf)

To reproduce this problem, do the following in an interactive VERDI session:

1.  Load (from _atmos_) this file: /work/MOD3DEV/jherwehe/verdi/test/CMAQ_v53_Dev/CMAQv53b3fix_2016_12US1_M3Dry_Bidi/07/CCTM_ACONC_v53_intel18.0_2016_CONUS_test_20160715.nc
2.  Select O3 (ppmV)[1] from the Variables list.
3.  In the Formulas tab, add 1000.*O3[1] to the formulas list (to convert the O3 units to ppbV to match the obs overlay units) and create a Tile Plot of this formula.
4.  Now load the following binary (netCDF/IOAPI) obs overlay file (also available on _atmos_): /work/MOD3DEV/jherwehe/verdi/test/wtt/obs_data/Release_Obs_Dev/Tools/Obs_Overlay/AQS_Hourly_PAVE_VERDI_overlay_2016.ncf
5.  Initially looks good, but now advance the Time Step and watch the obs sites disappear, then gradually reappear as you advance the time, but showing incorrect observation values.
6.  To illustrate that the initial time matching between the obs and model output is working, make a fresh Tile Plot of 1000.*O3[1], advance to some point in time (e.g., to 20UTC), then overlay the O3 obs again and notice that the agreement between the obs and the CMAQ O3 field looks much better than if you had advanced to 20UTC from the initial time.

To summarize, there seems to be three symptoms of the obs overlay problem:

1.  **Sites vanish**:  Obs sites disappear starting at 00UTC when looping in time, then gradually reappear.
2.  **Out of synch**:  Initially, obs values appear correct, but after any time changes, they lose synchronization with the model; i.e., you can never return  back to the initial time and have it display correctly.
3.  **Incorrect obs values**:  Also, after any time changes, the obs seem to show wrong values (i.e., less agreement with the model compared to just adding the obs overlay at a particular point in time; see the examples in the PDF provided above).

Problem items 2. and 3. above are probably closely related.  Problem item 1 is likely a separate flaw in the VERDI coding.

**Comment by jherwehe** [comment]
**[Updated test of this issue (#209) using the 20190803 version of VERDI v2.0beta on both Linux and Windows 10.]**

After discussing this overlay problem with Wyat Appel (@wkappel), we're convinced that VERDI is matching the IOAPI-formatted overlay with the hour of the CMAQ day, but not the date.  As reported previously, VERDI would match date and time on the initial plotting of the overlay, but lose track of the date if the plotted time was changed via the Time Step spinner.  The 20190803 VERDI release no longer matches the date on the initial display of the overlay, but reverts to the corresponding hour on January 1, 2016, when using the 2016 full-year overlay file.  Therefore @systemsgo was correct that there were only two sites reporting O3 data for 00 UTC January 1, 2016; Wyat explained that many states turn off some or all of their O3 monitors during winter (e.g., Georgia), plus many observation sites are offline for a few hours at the beginning of the year for calibration and such.  M3XTRACT was used to extract the July 15, 2016 (i.e., 2016197) observations from the annual overlay file, and when plotted over the July 15, 2016, CMAQ O3 field, the overlay sites are plentiful and have values much closer to those of the CMAQ field.  See the updated comparison plots for 02 UTC and 20 UTC O3 below (click a plot pair to view larger versions):

**-- CMAQ v5.3beta O3 for July 15, 2016 --**

![obs_overlay_problem_comparison_using_20190803_02UTC](https://user-images.githubusercontent.com/16904596/62649525-0aefea80-b923-11e9-95bf-eaf7bb9e9610.png)
![obs_overlay_problem_comparison_using_20190803_20UTC](https://user-images.githubusercontent.com/16904596/62649545-10e5cb80-b923-11e9-9662-e6bf1eb7dba2.png)

To reproduce this problem, use these IOAPI-formatted files from _atmos_:

**1. CMAQ output:**
/work/MOD3DEV/jherwehe/verdi/test/CMAQ_v53_Dev/CMAQv53b3fix_2016_12US1_M3Dry_Bidi/07/CCTM_ACONC_v53_intel18.0_2016_CONUS_test_20160715.nc
(Don't forget to multiply O3 by 1000. to convert from ppmV to ppbV to match the overlay units.)

**2. Full-year (2016) overlay file:**
/work/MOD3DEV/jherwehe/verdi/test/wtt/obs_data/Release_Obs_Dev/Tools/Obs_Overlay/AQS_Hourly_PAVE_VERDI_overlay_2016.ncf

**3. One-day (July 15, 2016) overlay file:**
/work/MOD3DEV/jherwehe/verdi/test/wtt/obs_data/Release_Obs_Dev/Tools/Obs_Overlay/AQS_Hourly_PAVE_VERDI_overlay_20160715.ncf

**In conclusion...**
**Bug to be fixed:**  VERDI needs to match _both_ date and time between the displayed AQ model field and the overlay file to be properly synchronized.

**Comment by jherwehe** [comment]
Retested this issue using the 20190820 version of VERDI_2.0_beta on Linux and Windows and this problem is now fixed; obs overlay time AND date are synchronized with the underlying tile plot and remain synchronized when changing the displayed time step.

Closing this issue.
